{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package averaged_perceptron_tagger to\n",
      "[nltk_data]     /home/ekansh/nltk_data...\n",
      "[nltk_data]   Package averaged_perceptron_tagger is already up-to-\n",
      "[nltk_data]       date!\n",
      "[nltk_data] Downloading package stopwords to /home/ekansh/nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n",
      "[nltk_data] Downloading package wordnet to /home/ekansh/nltk_data...\n",
      "[nltk_data]   Package wordnet is already up-to-date!\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import nltk\n",
    "import os\n",
    "import re\n",
    "import math\n",
    "import operator\n",
    "import json\n",
    "import tqdm\n",
    "\n",
    "from nltk.stem import WordNetLemmatizer\n",
    "from nltk.corpus import stopwords\n",
    "from nltk.tokenize import sent_tokenize,word_tokenize\n",
    "nltk.download('averaged_perceptron_tagger')\n",
    "nltk.download('stopwords')\n",
    "Stopwords = set(stopwords.words('english'))\n",
    "wordlemmatizer = WordNetLemmatizer()\n",
    "nltk.download('wordnet')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def lemmatize_words(words):\n",
    "    lemmatized_words = []\n",
    "    for word in words:\n",
    "       lemmatized_words.append(wordlemmatizer.lemmatize(word))\n",
    "    return lemmatized_words\n",
    "# def stem_words(words):\n",
    "#     stemmed_words = []\n",
    "#     for word in words:\n",
    "#        stemmed_words.append(stemmer.stem(word))\n",
    "    return stemmed_words\n",
    "def remove_special_characters(text):\n",
    "    regex = r'[^a-zA-Z0-9\\s]'\n",
    "    text = re.sub(regex,'',text)\n",
    "    return text\n",
    "def freq(words):\n",
    "    words = [word.lower() for word in words]\n",
    "    dict_freq = {}\n",
    "    words_unique = []\n",
    "    for word in words:\n",
    "       if word not in words_unique:\n",
    "           words_unique.append(word)\n",
    "    for word in words_unique:\n",
    "       dict_freq[word] = words.count(word)\n",
    "    return dict_freq\n",
    "def pos_tagging(text):\n",
    "    pos_tag = nltk.pos_tag(text.split())\n",
    "    pos_tagged_noun_verb = []\n",
    "    for word,tag in pos_tag:\n",
    "        if tag == \"NN\" or tag == \"NNP\" or tag == \"NNS\" or tag == \"VB\" or tag == \"VBD\" or tag == \"VBG\" or tag == \"VBN\" or tag == \"VBP\" or tag == \"VBZ\":\n",
    "             pos_tagged_noun_verb.append(word)\n",
    "    return pos_tagged_noun_verb\n",
    "def tf_score(word,sentence):\n",
    "    freq_sum = 0\n",
    "    word_frequency_in_sentence = 0\n",
    "    len_sentence = len(sentence)\n",
    "    for word_in_sentence in sentence.split():\n",
    "        if word == word_in_sentence:\n",
    "            word_frequency_in_sentence = word_frequency_in_sentence + 1\n",
    "    tf =  word_frequency_in_sentence/ len_sentence\n",
    "    return tf\n",
    "def idf_score(no_of_sentences,word,sentences):\n",
    "    no_of_sentence_containing_word = 0\n",
    "    for sentence in sentences:\n",
    "        sentence = remove_special_characters(str(sentence))\n",
    "        sentence = re.sub(r'\\d+', '', sentence)\n",
    "        sentence = sentence.split()\n",
    "        sentence = [word for word in sentence if word.lower() not in Stopwords and len(word)>1]\n",
    "        sentence = [word.lower() for word in sentence]\n",
    "        sentence = [wordlemmatizer.lemmatize(word) for word in sentence]\n",
    "        if word in sentence:\n",
    "            no_of_sentence_containing_word = no_of_sentence_containing_word + 1\n",
    "    idf = math.log10(no_of_sentences/no_of_sentence_containing_word)\n",
    "    return idf\n",
    "def tf_idf_score(tf,idf):\n",
    "    return tf*idf\n",
    "def word_tfidf(dict_freq,word,sentences,sentence):\n",
    "    word_tfidf = []\n",
    "    tf = tf_score(word,sentence)\n",
    "    idf = idf_score(len(sentences),word,sentences)\n",
    "    tf_idf = tf_idf_score(tf,idf)\n",
    "    return tf_idf\n",
    "def sentence_importance(sentence,dict_freq,sentences):\n",
    "     sentence_score = 0\n",
    "     sentence = remove_special_characters(str(sentence)) \n",
    "     sentence = re.sub(r'\\d+', '', sentence)\n",
    "     pos_tagged_sentence = [] \n",
    "     no_of_sentences = len(sentences)\n",
    "     pos_tagged_sentence = pos_tagging(sentence)\n",
    "     for word in pos_tagged_sentence:\n",
    "          if word.lower() not in Stopwords and word not in Stopwords and len(word)>1: \n",
    "                word = word.lower()\n",
    "                word = wordlemmatizer.lemmatize(word)\n",
    "                sentence_score = sentence_score + word_tfidf(dict_freq,word,sentences,sentence)\n",
    "     return sentence_score\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/241 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 1/241 [04:01<16:04:03, 241.02s/it]\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 63\u001b[0m\n\u001b[1;32m     61\u001b[0m ni \u001b[38;5;241m=\u001b[39m json\u001b[38;5;241m.\u001b[39mloads(i)\n\u001b[1;32m     62\u001b[0m text \u001b[38;5;241m=\u001b[39m ni[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124marticle\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m---> 63\u001b[0m stri \u001b[38;5;241m=\u001b[39m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtext\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     64\u001b[0m finalarr\u001b[38;5;241m.\u001b[39mappend(stri)\n",
      "Cell \u001b[0;32mIn[3], line 24\u001b[0m, in \u001b[0;36mfunc\u001b[0;34m(text)\u001b[0m\n\u001b[1;32m     19\u001b[0m \u001b[38;5;66;03m# print(len(tokenized_sentence))\u001b[39;00m\n\u001b[1;32m     20\u001b[0m \u001b[38;5;66;03m# print('\\n')\u001b[39;00m\n\u001b[1;32m     21\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m sent \u001b[38;5;129;01min\u001b[39;00m tokenized_sentence:\n\u001b[1;32m     22\u001b[0m     \u001b[38;5;66;03m# print(c)\u001b[39;00m\n\u001b[1;32m     23\u001b[0m     \u001b[38;5;66;03m# print('\\n')\u001b[39;00m\n\u001b[0;32m---> 24\u001b[0m     sentenceimp \u001b[38;5;241m=\u001b[39m \u001b[43msentence_importance\u001b[49m\u001b[43m(\u001b[49m\u001b[43msent\u001b[49m\u001b[43m,\u001b[49m\u001b[43mword_freq\u001b[49m\u001b[43m,\u001b[49m\u001b[43mtokenized_sentence\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     25\u001b[0m     sentence_with_importance[c] \u001b[38;5;241m=\u001b[39m sentenceimp\n\u001b[1;32m     26\u001b[0m     c \u001b[38;5;241m=\u001b[39m c\u001b[38;5;241m+\u001b[39m\u001b[38;5;241m1\u001b[39m\n",
      "Cell \u001b[0;32mIn[2], line 73\u001b[0m, in \u001b[0;36msentence_importance\u001b[0;34m(sentence, dict_freq, sentences)\u001b[0m\n\u001b[1;32m     71\u001b[0m            word \u001b[38;5;241m=\u001b[39m word\u001b[38;5;241m.\u001b[39mlower()\n\u001b[1;32m     72\u001b[0m            word \u001b[38;5;241m=\u001b[39m wordlemmatizer\u001b[38;5;241m.\u001b[39mlemmatize(word)\n\u001b[0;32m---> 73\u001b[0m            sentence_score \u001b[38;5;241m=\u001b[39m sentence_score \u001b[38;5;241m+\u001b[39m \u001b[43mword_tfidf\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdict_freq\u001b[49m\u001b[43m,\u001b[49m\u001b[43mword\u001b[49m\u001b[43m,\u001b[49m\u001b[43msentences\u001b[49m\u001b[43m,\u001b[49m\u001b[43msentence\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     74\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m sentence_score\n",
      "Cell \u001b[0;32mIn[2], line 59\u001b[0m, in \u001b[0;36mword_tfidf\u001b[0;34m(dict_freq, word, sentences, sentence)\u001b[0m\n\u001b[1;32m     57\u001b[0m word_tfidf \u001b[38;5;241m=\u001b[39m []\n\u001b[1;32m     58\u001b[0m tf \u001b[38;5;241m=\u001b[39m tf_score(word,sentence)\n\u001b[0;32m---> 59\u001b[0m idf \u001b[38;5;241m=\u001b[39m \u001b[43midf_score\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43msentences\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43mword\u001b[49m\u001b[43m,\u001b[49m\u001b[43msentences\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     60\u001b[0m tf_idf \u001b[38;5;241m=\u001b[39m tf_idf_score(tf,idf)\n\u001b[1;32m     61\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m tf_idf\n",
      "Cell \u001b[0;32mIn[2], line 49\u001b[0m, in \u001b[0;36midf_score\u001b[0;34m(no_of_sentences, word, sentences)\u001b[0m\n\u001b[1;32m     47\u001b[0m sentence \u001b[38;5;241m=\u001b[39m [word \u001b[38;5;28;01mfor\u001b[39;00m word \u001b[38;5;129;01min\u001b[39;00m sentence \u001b[38;5;28;01mif\u001b[39;00m word\u001b[38;5;241m.\u001b[39mlower() \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m Stopwords \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(word)\u001b[38;5;241m>\u001b[39m\u001b[38;5;241m1\u001b[39m]\n\u001b[1;32m     48\u001b[0m sentence \u001b[38;5;241m=\u001b[39m [word\u001b[38;5;241m.\u001b[39mlower() \u001b[38;5;28;01mfor\u001b[39;00m word \u001b[38;5;129;01min\u001b[39;00m sentence]\n\u001b[0;32m---> 49\u001b[0m sentence \u001b[38;5;241m=\u001b[39m \u001b[43m[\u001b[49m\u001b[43mwordlemmatizer\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mlemmatize\u001b[49m\u001b[43m(\u001b[49m\u001b[43mword\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mword\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43msentence\u001b[49m\u001b[43m]\u001b[49m\n\u001b[1;32m     50\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m word \u001b[38;5;129;01min\u001b[39;00m sentence:\n\u001b[1;32m     51\u001b[0m     no_of_sentence_containing_word \u001b[38;5;241m=\u001b[39m no_of_sentence_containing_word \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m1\u001b[39m\n",
      "Cell \u001b[0;32mIn[2], line 49\u001b[0m, in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m     47\u001b[0m sentence \u001b[38;5;241m=\u001b[39m [word \u001b[38;5;28;01mfor\u001b[39;00m word \u001b[38;5;129;01min\u001b[39;00m sentence \u001b[38;5;28;01mif\u001b[39;00m word\u001b[38;5;241m.\u001b[39mlower() \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m Stopwords \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(word)\u001b[38;5;241m>\u001b[39m\u001b[38;5;241m1\u001b[39m]\n\u001b[1;32m     48\u001b[0m sentence \u001b[38;5;241m=\u001b[39m [word\u001b[38;5;241m.\u001b[39mlower() \u001b[38;5;28;01mfor\u001b[39;00m word \u001b[38;5;129;01min\u001b[39;00m sentence]\n\u001b[0;32m---> 49\u001b[0m sentence \u001b[38;5;241m=\u001b[39m [\u001b[43mwordlemmatizer\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mlemmatize\u001b[49m\u001b[43m(\u001b[49m\u001b[43mword\u001b[49m\u001b[43m)\u001b[49m \u001b[38;5;28;01mfor\u001b[39;00m word \u001b[38;5;129;01min\u001b[39;00m sentence]\n\u001b[1;32m     50\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m word \u001b[38;5;129;01min\u001b[39;00m sentence:\n\u001b[1;32m     51\u001b[0m     no_of_sentence_containing_word \u001b[38;5;241m=\u001b[39m no_of_sentence_containing_word \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m1\u001b[39m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/nltk/stem/wordnet.py:45\u001b[0m, in \u001b[0;36mWordNetLemmatizer.lemmatize\u001b[0;34m(self, word, pos)\u001b[0m\n\u001b[1;32m     33\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mlemmatize\u001b[39m(\u001b[38;5;28mself\u001b[39m, word: \u001b[38;5;28mstr\u001b[39m, pos: \u001b[38;5;28mstr\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mn\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m \u001b[38;5;28mstr\u001b[39m:\n\u001b[1;32m     34\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Lemmatize `word` using WordNet's built-in morphy function.\u001b[39;00m\n\u001b[1;32m     35\u001b[0m \u001b[38;5;124;03m    Returns the input word unchanged if it cannot be found in WordNet.\u001b[39;00m\n\u001b[1;32m     36\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     43\u001b[0m \u001b[38;5;124;03m    :return: The lemma of `word`, for the given `pos`.\u001b[39;00m\n\u001b[1;32m     44\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m---> 45\u001b[0m     lemmas \u001b[38;5;241m=\u001b[39m \u001b[43mwn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_morphy\u001b[49m\u001b[43m(\u001b[49m\u001b[43mword\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpos\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     46\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mmin\u001b[39m(lemmas, key\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mlen\u001b[39m) \u001b[38;5;28;01mif\u001b[39;00m lemmas \u001b[38;5;28;01melse\u001b[39;00m word\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/nltk/corpus/reader/wordnet.py:2100\u001b[0m, in \u001b[0;36mWordNetCorpusReader._morphy\u001b[0;34m(self, form, pos, check_exceptions)\u001b[0m\n\u001b[1;32m   2097\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m filter_forms([form] \u001b[38;5;241m+\u001b[39m exceptions[form])\n\u001b[1;32m   2099\u001b[0m \u001b[38;5;66;03m# 1. Apply rules once to the input to get y1, y2, y3, etc.\u001b[39;00m\n\u001b[0;32m-> 2100\u001b[0m forms \u001b[38;5;241m=\u001b[39m \u001b[43mapply_rules\u001b[49m\u001b[43m(\u001b[49m\u001b[43m[\u001b[49m\u001b[43mform\u001b[49m\u001b[43m]\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2102\u001b[0m \u001b[38;5;66;03m# 2. Return all that are in the database (and check the original too)\u001b[39;00m\n\u001b[1;32m   2103\u001b[0m results \u001b[38;5;241m=\u001b[39m filter_forms([form] \u001b[38;5;241m+\u001b[39m forms)\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/nltk/corpus/reader/wordnet.py:2076\u001b[0m, in \u001b[0;36mWordNetCorpusReader._morphy.<locals>.apply_rules\u001b[0;34m(forms)\u001b[0m\n\u001b[1;32m   2075\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mapply_rules\u001b[39m(forms):\n\u001b[0;32m-> 2076\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m[\u001b[49m\n\u001b[1;32m   2077\u001b[0m \u001b[43m        \u001b[49m\u001b[43mform\u001b[49m\u001b[43m[\u001b[49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m-\u001b[39;49m\u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mold\u001b[49m\u001b[43m)\u001b[49m\u001b[43m]\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mnew\u001b[49m\n\u001b[1;32m   2078\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mform\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mforms\u001b[49m\n\u001b[1;32m   2079\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mold\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnew\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43msubstitutions\u001b[49m\n\u001b[1;32m   2080\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43;01mif\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mform\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mendswith\u001b[49m\u001b[43m(\u001b[49m\u001b[43mold\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2081\u001b[0m \u001b[43m    \u001b[49m\u001b[43m]\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/nltk/corpus/reader/wordnet.py:2080\u001b[0m, in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m   2075\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mapply_rules\u001b[39m(forms):\n\u001b[1;32m   2076\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m [\n\u001b[1;32m   2077\u001b[0m         form[: \u001b[38;5;241m-\u001b[39m\u001b[38;5;28mlen\u001b[39m(old)] \u001b[38;5;241m+\u001b[39m new\n\u001b[1;32m   2078\u001b[0m         \u001b[38;5;28;01mfor\u001b[39;00m form \u001b[38;5;129;01min\u001b[39;00m forms\n\u001b[1;32m   2079\u001b[0m         \u001b[38;5;28;01mfor\u001b[39;00m old, new \u001b[38;5;129;01min\u001b[39;00m substitutions\n\u001b[0;32m-> 2080\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[43mform\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mendswith\u001b[49m\u001b[43m(\u001b[49m\u001b[43mold\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2081\u001b[0m     ]\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# writefile = open('summary.txt', 'w')       \n",
    "\n",
    "def func (text):\n",
    "    tokenized_sentence = sent_tokenize(text)\n",
    "    text = remove_special_characters(str(text))\n",
    "    text = re.sub(r'\\d+', '', text)\n",
    "    tokenized_words_with_stopwords = word_tokenize(text)\n",
    "    tokenized_words = [word for word in tokenized_words_with_stopwords if word not in Stopwords]\n",
    "    tokenized_words = [word for word in tokenized_words if len(word) > 1]\n",
    "    tokenized_words = [word.lower() for word in tokenized_words]\n",
    "    tokenized_words = lemmatize_words(tokenized_words)\n",
    "    word_freq = freq(tokenized_words)\n",
    "    # input_user = int(input('Percentage of information to retain(in percent):'))\n",
    "    # no_of_sentences = int((input_user * len(tokenized_sentence))/100)\n",
    "    no_of_sentences = 1\n",
    "    # print(no_of_sentences)\n",
    "    c = 1\n",
    "    sentence_with_importance = {}\n",
    "    # print(len(tokenized_sentence))\n",
    "    # print('\\n')\n",
    "    for sent in tokenized_sentence:\n",
    "        # print(c)\n",
    "        # print('\\n')\n",
    "        sentenceimp = sentence_importance(sent,word_freq,tokenized_sentence)\n",
    "        sentence_with_importance[c] = sentenceimp\n",
    "        c = c+1\n",
    "    # print(c)\n",
    "    sentence_with_importance = sorted(sentence_with_importance.items(), key=operator.itemgetter(1),reverse=True)\n",
    "    cnt = 0\n",
    "    summary = []\n",
    "    sentence_no = []\n",
    "    # print(cnt)\n",
    "\n",
    "    for word_prob in sentence_with_importance:\n",
    "        if cnt < no_of_sentences:\n",
    "            sentence_no.append(word_prob[0])\n",
    "            cnt = cnt+1\n",
    "        else:\n",
    "            break\n",
    "    sentence_no.sort()\n",
    "    cnt = 1\n",
    "    for sentence in tokenized_sentence:\n",
    "        if cnt in sentence_no:\n",
    "            summary.append(sentence)\n",
    "        cnt = cnt+1\n",
    "    summary = \" \".join(summary)\n",
    "    # print(\"\\n\")\n",
    "    # print(\"Summary:\")\n",
    "    # print(summary)\n",
    "    return summary\n",
    "\n",
    "\n",
    "# read the json file\n",
    "with open('testingpurpose.jsonl') as f:\n",
    "    data = f.readlines()\n",
    "    # print(data)\n",
    "\n",
    "    finalarr = []\n",
    "\n",
    "    for i in tqdm.tqdm(data):\n",
    "        ni = json.loads(i)\n",
    "        text = ni['article']\n",
    "        stri = func(text)\n",
    "        finalarr.append(stri)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['5×10–5 bar–1 ) .', 'For example , wrongly rejecting a food item might see an animal missing a meal , but wrongly accepting a food item could see an animal ingesting poison .', 'ac .', 'Hosts were only included in these analyses if they had >35 samples in either the juvenile and prime age class or prime age and older adult class ( no host had >35 samples in all three age classes; 13 hosts were included in the juvenile and matched prime age adult groups; a separate set of 13 hosts were included in the older adult and matched prime age adult groups ) .', 'doi .', '25 ng/µl .', 'ilmn12 .', ', chr17q21 locus ) .', 'Fertilizers were applied three times per year , namely once just before sowing ( 50 kg/ha N , 85 kg/ha P , 50 kg/ha K ) , once when wheat was at the tillering stage ( 50 kg/ha N , 85 kg/ha P , 50 kg/ha K ) , and once when wheat was flowering ( 20 kg/ha N , 34 kg/ha P , 20 kg/ha K ) .', 'Then two choice targets appeared at 1 of 3 different pairs of positions ( 140 mm to the left and right of the central fixation target and either at the same height as the central fixation target or 35 mm below or 35 mm above it ) .', 'at room temp .', ', 2009; van Geel et al .', '5 ) at a dose of 35 mg/kg for day-1 , and 35 mg/kg for day-2 at the 5th week .', 'Turn angle ( α ) : the angle between the line passing through the CoM and the tip of the snout at the onset of stage 1 , and the line passing through the CoM at the onset of stage 1 and the CoM at the end of stage 2 .', 'io .', '077 g/ml , cat .', '825 Å/pixel .', 'org ) .', '5 mm .', 'Figure 2 , Figure 2—source data 1 , Figure 2—source data 2 , Figure 2—source data 3 , Figure 2—source data 4 , Figure 3 , Figure 3—source data 1 , Figure 3—source data 2 , Figure 3—source data 3 , and Figure 3—source data 4 show the summarized effect estimates .', '1 mL/min on the following gradient: 0% B for 0–2 min , linear increase 0–35% B in 2–60 min , 35–85% B in 60–62 min , maintain at 85% B until 68 min , linear decrease to 0% in 68–70 min , and finally equilibrated the system at 0% B until 85 min .', '2 mg/ml k-casein ) .', 'For mapping between euchromatin and heterochromatin ( Figure 3B ) , the following limits of euchromatin were used: chr = \"2L\" & start <21400000; chr = \"2R\" & start >5970000; chr = \"3L\" & start <22270000 chr = \"3R\" & start >4191000; chr == \"X\" & start <21400000 75 b .', 'ed .', 'edu ) .', 'CD11B+ gating quantifies the myeloid compartment .', 'msu .', '0 eq ) .', 'pct=0 .', 'mulitqtl .', '5 µm step size .', 'jsp ) .', 'ac .', '4th , 5th , etc . )', 'ac .', 'After a 10 min wash with WB transfer buffer supplied with 1 mM EDTA , followed by a 10 min wash with transfer buffer without EDTA .', 'io; probability > 0 .', '2% Triton X-100 and 1% BSA ) , using the following antibodies and dilutions: anti-PDGFRβ/CD140b ( dilution 1:200 , clone APB5 , eBioscience ) +anti rat AF647 ( dilution 1:200 , Jackson IR ) ; anti-CD11c AF647 , ( dilution 1:100 , clone N418 , Biolegend ) ; anti-GFP ( dilution 1:100 , rabbit polyclonal , A11122 Thermo Fisher Scientific ) +anti-rabbit AF 488 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-CD31/PECAM-1 ( dilution 1:300 , goat polyclonal , AF3628 R&D Systems ) +anti goat AF546 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-Heparan Sulfate ( dilution 1:50 , clone F58-10E4 , Amsbio ) +anti IgM conjugated to APC ( dilution 100 , clone II/41 , Thermo Fisher Scientific ) ; anti-LYVE-1 ( dilution 1:300 , rabbit polyclonal , Novus Biologicals ) +anti-rabbit AF647 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-Collagen type IV ( dilution 1:100 , rabbit polyclonal , Abcam ) +anti-rabbit AF488 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-Laminin ( dilution 1:300 , rabbit polyclonal , Sigma-Aldrich ) +anti-rabbit AF488 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-CD54/ICAM1 ( dilution 1:100 , clone YN1/1 .', 'vcf .', '5 mm ) .', '89 log 10 copies/ml , p=0 .', 'dna .', 'Among the 18 infants , 11 performed newborn stepping , 15 performed stepping at 3 months old , 15 performed newborn kicking , 15 performed kicking at 3 months old , and 15 performed walking as toddlers ( Table 2 ) .', 'ox .', '1\\u2009g bovine serum albumin .', 'org/ ) .', 'org/ ) .', 'ac .', 'Optimal E was selected by measuring TE as follows: ( 2 ) TE=1T∑t=1Tlog\\u2061 ( p ( xt+tp\\xa0|\\xa0yt , xt , xt−τ , ⋯ , xt− ( E−1 ) τ ) p ( xt+tp\\xa0|\\xa0yt , xt , xt−τ , ⋯ , xt− ( ER−1 ) τ ) ) , where ER ( <E ) is the optimal embedding dimension of lower dimensional models .', ', 2019 , table 1 ) .', '815 mm/pixel ) .', '2 pixel .', '( 17 ) cos\\u2061 ( 2β ) = ( a~2−b~2a~2+b~2 ) 1+25η−G′′/ωη1K ( a~2+b~22a~b~ ) 21+25η−G′′/ωη1K ( a~2−b~22a~b~ ) 2 ( corresponding to Equation 80 in Roscoe , 1967 ) .', 'mit .', '4% Agar gel with the WT allele producing a 1868 bp amplicon and the floxed allele producing a 2142 bp amplicon .', 'com/ ) .', 'coli .', '05 cutoff .', 'py .', ', politics , health , science , etc . )', '7 nm .', 'nlm .', '5 kV ) mode electrospray ionization ( capillary temperature 300 °C; sheath gas 50; Aux gas 20; sweep gas 2; probe temp 120 °C ) .', 'twofolds of 1% would remarkably differ from twofolds of 10% ) .', 'nitrc .', 'ucd .', 'org/10 .', ', 2004 ) ; tissue 1 = cortical grey matter , tissue 2 = sub-cortical grey matter , tissue 3 = white matter , tissue 4 = CSF , and tissue 5 = pathological tissue .', '1% Thioflavin-S in 70% ethanol 25 min , rinsed in 80% ethanol for 5 min and in distilled water for 5 min .', 'We used the primary antibodies: anti-CD44 ( clone IM7 ) , anti-MHC class I ( clone 28-8-6 ) ( both from BioLegend ) , anti-β-catenin ( clone D2U8Y ) , anti-phosphorylated STAT3 ( clone D3A7 ) , anti-active integrin 1β ( polyclonal ) , anti-Granzyme B ( clone 5A1E ) ( all from Cell Signaling Technology ) , anti-integrin 1β ( clone 9EG7 ) , anti-E-cadherin ( clone 36 ) ( both from BD Biosciences ) , anti-TRP2 ( clone C-9 , Santa Cruz Biotechnology ) , anti-gp100 ( clone EP4863 Galon et al .', '001 , ηp2p2 .', 'org/10 .', '4 ) , and cut into multiple 1 mm × 1 mm strips in fluid; several strips were simultaneously cut into 1 mm × 1 mm × 1 mm blocks for TEM .', '/ml ) .', '2 cm2; slice number = 20; slice thickness = 1 mm; repetition number = 6 .', 'et al .', 'org/10 .', 'Enrichment analysis identified Gene Ontology ( GO ) biological processes linked to the cardiovascular system including striated muscle cell development , muscle cell development , muscle system process , cardiac muscle contraction , heart contraction , heart process , contraction muscle cell development , and muscle tissue progress among others ( Figure 6b ) .', 'The surface function of Imaris was used to segment the wing pouch and the surface volume was calculated by the software .', 'yield ( i .', 'fa .', 'Time-resolved oxygen polarography was performed using a custom-made centrifugable static ring-disk electrode assembly of a bare platinum working electrode and silver-ring counter electrode , as previously described ( Dilbeck et al .', 'nlm .', 'msu .', 'Sequencing was performed using HiSeq 4000 ( Illumina ) at IGM Genomics Center , University of California , San Diego , with the following read length: Read 1 , 28 bp , including 16 bp cell barcode and 12 bp unique molecular identifier ( UMI ) ; Read 2 , 98 bp transcript insert; i7 sample index , 8 bp .', 'uth .', 'The ion source settings were capillary voltage , 3500 V; nozzle voltage , 2000 V; fragmentor voltage , 180 V; drying gas and sheath gas temperature , 350°C; drying gas flow , 10 L/min; sheath gas flow , 11 L/min; nebulizer pressure , 60 lb/in2 .', '5 nm .', 'Various companies now sell personalised diets based on nutrigenomics or personalised exercise plans based on sportomics; others promise personalised learning approaches based on behavioural genetics , to name a few .', 'In a separate set of experiments , we tracked vsx1+ sister neurons from 48 to 96 hpf by embedding fish for imaging at 48 hpf , freeing from agarose after imaging , and re-embedding at 96 hpf .', 'med .', 'nlm .', 'They are identified as ( i ) the Gaussian variability in response around the target value ( target response proportion ) , ( ii ) Gaussian variability in response around the non-target value , that is , mistakenly reporting feature of other presented items ( swap error ) , and ( iii ) random responses ( uniform response proportion ) ( Bays et al .', ', 1987 ) , rabbit anti-CD4 ( Novus Biologicals , 1:500 ) , mouse anti-REPO ( Developmental Studies Hybridoma Bank 8D12 , 1:5 ) , rabbit anti-DsRed ( Clontech #632496 , 1:200 ) , rat anti-HA ( Roche 3F10 , 1:100 ) , mouse anti-CaMPARI-red ( Absolute antibody 4F6 , 1:1000 ) , goat anti-HRP-Cy3 ( Jackson ImmunoResearch , 1:100 ) , goat anti-rat Alexa488 ( Invitrogen , 1:500 ) , goat anti-chicken Alexa488 ( Invitrogen , 1:500 ) , goat anti-mouse Alexa546 ( Invitrogen , 1:500 ) , goat anti-rabbit Alexa546 ( Invitrogen , 1:500 ) , goat anti-rat Alexa633 ( Invitrogen , 1:500 ) , and goat anti-rabbit Alexa633 ( Invitrogen , 1:500 ) .', 'ac .', 'The CCG at time lag τ is the correlation between the velocities of the target vt ( target ) at time t and the velocity of the response vt−τ ( response ) at time ( t−τ ) : ( 9 ) ρ ( τ ) =cov ( vt ( target ) , vt−τ ( response ) ) s ( vt ( target ) ) ⋅s ( vt−τ ( response ) ) , where cov\\u2062 ( x , y ) and s\\u2062 ( x ) are the covariance and standard deviation across all time steps and trials .', 'Performing the change of variables using the derivative dξn+1dpn+1=σ22pn+1μ−2pn+12μ>0 yields the transfer function ( 16 ) fp ( pn+1|pn ) =12μpn+1 ( 1−pn+1 ) 2πδtσ[pnexp\\u2061{−12σ2δt ( σ22μln\\u2061 ( pn−1 ) pn+1pn ( pn+1−1 ) −δtμ ) 2}+ ( 1−pn ) exp\\u2061{−12σ2δt ( σ22μln\\u2061 ( pn−1 ) pn+1pn ( pn+1−1 ) +δtμ ) 2}] .', 'The sediment samples were extracted using 5 ml of hexane: acetone ( 3:1 v/v ) , vortexed for 5 min , followed by ultrasonication for 15 min and centrifugation for 3 min at 5000 rpm .', 'org/10 .', 'ac .', '6 cm ) .', 'Standard rodent diet and acidified tap water were provided ad libitum .', ', 2014; van der Hoorn et al .', 'org/ ) .', '1 260/280 nm ratio .', 'dna .', '4 mm voxel size , echo time ( TE ) = 20 ms , repetition time ( TR ) = 1 .', 'nves-k !', 'nov .', 'org/ ) .', '01 ml/g ) .', 'zip .', 'We then modeled the bladder filling and voiding over a 6 hr duty cycle with different states: filling bladder ( 0 to 2 hr , 0% to 10% strain ) , filled bladder ( 2 to 4 hr , 10% strain ) , voiding bladder over a period of 2 min ( 4 hr to 4:02 hr , 10% to 0% strain ) , and voided bladder ( 4:02 hr to 6 hr , 0% strain ) ( Figure 1D ) .', 'ac .', 'nlm .', 'The VPA treatment group received 200 mg/kg VPA in 15% DMSO and 85% saline , whereas the vehicle control group received the same volume of 15% DMSO and 85% saline without VPA .', ', 2011 ) , plated on LB agar containing ampicillin ( 100 µg/ml ) and kanamycin ( 50 µg/ml ) , and incubated for 16 hr at 37°C .', '7] v[+t1 .', 'com/ .', 'ed .', '2% glucose .', '6% w/v ) .', 'soe .', 'Blots were washed three times , 5 min each in PBST at room temperature with agitation before being incubated in 1:100 LARP1 mouse monoclonal ( Santa Cruz ) primary antibody in 2% BSA ( Research Products International A30075 ) in PBST overnight at 4 °C with agitation with 1:5000 beta actin mouse monoclonal antibody ( Sigma A5441 ) in 2% milk powder in PBST for 2 hr at room temperature with agitation or with 1:10 , 000 H3 mouse monoclonal antibody ( Abcam ab10799 ) in 2% milk powder in PBST for 2 hr at room temperature with agitation .', 'org/10 .', '5 mg/kg docetaxel ( days 7 , 9 , 11 post-implantation ) and 1 dose of 10 mg/kg docetaxel ( day 14 post-implantation ) .', 'nlm .', 'com ) .', 'io .', 'org ) .', 'ac .', '2% , odds ratio = 5 .', '0076 ) , albeit weaker ( R2=0 .', 'New experimental condition terms were developed with a prefix of ‘delivery mechanism’: for example , ‘delivery mechanism: agrobacterium , ’ ‘delivery mechanism: heterologous organism , ’ and ‘delivery mechanism: pathogen inoculation .', '5 ng DNA .', 'cbu .', 'SD1 and SD2 form the outer domain , and SD3 and SD4 form the inner domain ( Figure 1 ) .', 'umn .', 'coli cell envelope .', 'csc .', 'To specify the Bayesian logistic regression ( BLR ) models for the probability of owning a mobile phone , we modeled phone ownership yij of individual i ∈ ( 1 , nj ) in country j ∈ ( 1 , 33 ) as a Bernoulli variable with probability θij : ( 1 ) yij|θij∼Bern ( θij ) The probability of owning a mobile phone , P ( yij=1 ) =θij , was then modeled using the logit-link function: ( 2 ) logit ( θij ) =β0+β1x1ij where β0 is the population-level intercept , and β1 is a regression coefficient that quantifies the influence of predictor variable x1ij ; we used a separate BLR model for each of the five variables .', 'ac .']\n"
     ]
    }
   ],
   "source": [
    "print(finalarr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5×10–5 bar–1 ) .\n",
      "For example , wrongly rejecting a food item might see an animal missing a meal , but wrongly accepting a food item could see an animal ingesting poison .\n",
      "ac .\n",
      "Hosts were only included in these analyses if they had >35 samples in either the juvenile and prime age class or prime age and older adult class ( no host had >35 samples in all three age classes; 13 hosts were included in the juvenile and matched prime age adult groups; a separate set of 13 hosts were included in the older adult and matched prime age adult groups ) .\n",
      "doi .\n",
      "25 ng/µl .\n",
      "ilmn12 .\n",
      ", chr17q21 locus ) .\n",
      "Fertilizers were applied three times per year , namely once just before sowing ( 50 kg/ha N , 85 kg/ha P , 50 kg/ha K ) , once when wheat was at the tillering stage ( 50 kg/ha N , 85 kg/ha P , 50 kg/ha K ) , and once when wheat was flowering ( 20 kg/ha N , 34 kg/ha P , 20 kg/ha K ) .\n",
      "Then two choice targets appeared at 1 of 3 different pairs of positions ( 140 mm to the left and right of the central fixation target and either at the same height as the central fixation target or 35 mm below or 35 mm above it ) .\n",
      "at room temp .\n",
      ", 2009; van Geel et al .\n",
      "5 ) at a dose of 35 mg/kg for day-1 , and 35 mg/kg for day-2 at the 5th week .\n",
      "Turn angle ( α ) : the angle between the line passing through the CoM and the tip of the snout at the onset of stage 1 , and the line passing through the CoM at the onset of stage 1 and the CoM at the end of stage 2 .\n",
      "io .\n",
      "077 g/ml , cat .\n",
      "825 Å/pixel .\n",
      "org ) .\n",
      "5 mm .\n",
      "Figure 2 , Figure 2—source data 1 , Figure 2—source data 2 , Figure 2—source data 3 , Figure 2—source data 4 , Figure 3 , Figure 3—source data 1 , Figure 3—source data 2 , Figure 3—source data 3 , and Figure 3—source data 4 show the summarized effect estimates .\n",
      "1 mL/min on the following gradient: 0% B for 0–2 min , linear increase 0–35% B in 2–60 min , 35–85% B in 60–62 min , maintain at 85% B until 68 min , linear decrease to 0% in 68–70 min , and finally equilibrated the system at 0% B until 85 min .\n",
      "2 mg/ml k-casein ) .\n",
      "For mapping between euchromatin and heterochromatin ( Figure 3B ) , the following limits of euchromatin were used: chr = \"2L\" & start <21400000; chr = \"2R\" & start >5970000; chr = \"3L\" & start <22270000 chr = \"3R\" & start >4191000; chr == \"X\" & start <21400000 75 b .\n",
      "ed .\n",
      "edu ) .\n",
      "CD11B+ gating quantifies the myeloid compartment .\n",
      "msu .\n",
      "0 eq ) .\n",
      "pct=0 .\n",
      "mulitqtl .\n",
      "5 µm step size .\n",
      "jsp ) .\n",
      "ac .\n",
      "4th , 5th , etc . )\n",
      "ac .\n",
      "After a 10 min wash with WB transfer buffer supplied with 1 mM EDTA , followed by a 10 min wash with transfer buffer without EDTA .\n",
      "io; probability > 0 .\n",
      "2% Triton X-100 and 1% BSA ) , using the following antibodies and dilutions: anti-PDGFRβ/CD140b ( dilution 1:200 , clone APB5 , eBioscience ) +anti rat AF647 ( dilution 1:200 , Jackson IR ) ; anti-CD11c AF647 , ( dilution 1:100 , clone N418 , Biolegend ) ; anti-GFP ( dilution 1:100 , rabbit polyclonal , A11122 Thermo Fisher Scientific ) +anti-rabbit AF 488 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-CD31/PECAM-1 ( dilution 1:300 , goat polyclonal , AF3628 R&D Systems ) +anti goat AF546 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-Heparan Sulfate ( dilution 1:50 , clone F58-10E4 , Amsbio ) +anti IgM conjugated to APC ( dilution 100 , clone II/41 , Thermo Fisher Scientific ) ; anti-LYVE-1 ( dilution 1:300 , rabbit polyclonal , Novus Biologicals ) +anti-rabbit AF647 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-Collagen type IV ( dilution 1:100 , rabbit polyclonal , Abcam ) +anti-rabbit AF488 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-Laminin ( dilution 1:300 , rabbit polyclonal , Sigma-Aldrich ) +anti-rabbit AF488 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-CD54/ICAM1 ( dilution 1:100 , clone YN1/1 .\n",
      "vcf .\n",
      "5 mm ) .\n",
      "89 log 10 copies/ml , p=0 .\n",
      "dna .\n",
      "Among the 18 infants , 11 performed newborn stepping , 15 performed stepping at 3 months old , 15 performed newborn kicking , 15 performed kicking at 3 months old , and 15 performed walking as toddlers ( Table 2 ) .\n",
      "ox .\n",
      "1 g bovine serum albumin .\n",
      "org/ ) .\n",
      "org/ ) .\n",
      "ac .\n",
      "Optimal E was selected by measuring TE as follows: ( 2 ) TE=1T∑t=1Tlog⁡ ( p ( xt+tp | yt , xt , xt−τ , ⋯ , xt− ( E−1 ) τ ) p ( xt+tp | yt , xt , xt−τ , ⋯ , xt− ( ER−1 ) τ ) ) , where ER ( <E ) is the optimal embedding dimension of lower dimensional models .\n",
      ", 2019 , table 1 ) .\n",
      "815 mm/pixel ) .\n",
      "2 pixel .\n",
      "( 17 ) cos⁡ ( 2β ) = ( a~2−b~2a~2+b~2 ) 1+25η−G′′/ωη1K ( a~2+b~22a~b~ ) 21+25η−G′′/ωη1K ( a~2−b~22a~b~ ) 2 ( corresponding to Equation 80 in Roscoe , 1967 ) .\n",
      "mit .\n",
      "4% Agar gel with the WT allele producing a 1868 bp amplicon and the floxed allele producing a 2142 bp amplicon .\n",
      "com/ ) .\n",
      "coli .\n",
      "05 cutoff .\n",
      "py .\n",
      ", politics , health , science , etc . )\n",
      "7 nm .\n",
      "nlm .\n",
      "5 kV ) mode electrospray ionization ( capillary temperature 300 °C; sheath gas 50; Aux gas 20; sweep gas 2; probe temp 120 °C ) .\n",
      "twofolds of 1% would remarkably differ from twofolds of 10% ) .\n",
      "nitrc .\n",
      "ucd .\n",
      "org/10 .\n",
      ", 2004 ) ; tissue 1 = cortical grey matter , tissue 2 = sub-cortical grey matter , tissue 3 = white matter , tissue 4 = CSF , and tissue 5 = pathological tissue .\n",
      "1% Thioflavin-S in 70% ethanol 25 min , rinsed in 80% ethanol for 5 min and in distilled water for 5 min .\n",
      "We used the primary antibodies: anti-CD44 ( clone IM7 ) , anti-MHC class I ( clone 28-8-6 ) ( both from BioLegend ) , anti-β-catenin ( clone D2U8Y ) , anti-phosphorylated STAT3 ( clone D3A7 ) , anti-active integrin 1β ( polyclonal ) , anti-Granzyme B ( clone 5A1E ) ( all from Cell Signaling Technology ) , anti-integrin 1β ( clone 9EG7 ) , anti-E-cadherin ( clone 36 ) ( both from BD Biosciences ) , anti-TRP2 ( clone C-9 , Santa Cruz Biotechnology ) , anti-gp100 ( clone EP4863 Galon et al .\n",
      "001 , ηp2p2 .\n",
      "org/10 .\n",
      "4 ) , and cut into multiple 1 mm × 1 mm strips in fluid; several strips were simultaneously cut into 1 mm × 1 mm × 1 mm blocks for TEM .\n",
      "/ml ) .\n",
      "2 cm2; slice number = 20; slice thickness = 1 mm; repetition number = 6 .\n",
      "et al .\n",
      "org/10 .\n",
      "Enrichment analysis identified Gene Ontology ( GO ) biological processes linked to the cardiovascular system including striated muscle cell development , muscle cell development , muscle system process , cardiac muscle contraction , heart contraction , heart process , contraction muscle cell development , and muscle tissue progress among others ( Figure 6b ) .\n",
      "The surface function of Imaris was used to segment the wing pouch and the surface volume was calculated by the software .\n",
      "yield ( i .\n",
      "fa .\n",
      "Time-resolved oxygen polarography was performed using a custom-made centrifugable static ring-disk electrode assembly of a bare platinum working electrode and silver-ring counter electrode , as previously described ( Dilbeck et al .\n",
      "nlm .\n",
      "msu .\n",
      "Sequencing was performed using HiSeq 4000 ( Illumina ) at IGM Genomics Center , University of California , San Diego , with the following read length: Read 1 , 28 bp , including 16 bp cell barcode and 12 bp unique molecular identifier ( UMI ) ; Read 2 , 98 bp transcript insert; i7 sample index , 8 bp .\n",
      "uth .\n",
      "The ion source settings were capillary voltage , 3500 V; nozzle voltage , 2000 V; fragmentor voltage , 180 V; drying gas and sheath gas temperature , 350°C; drying gas flow , 10 L/min; sheath gas flow , 11 L/min; nebulizer pressure , 60 lb/in2 .\n",
      "5 nm .\n",
      "Various companies now sell personalised diets based on nutrigenomics or personalised exercise plans based on sportomics; others promise personalised learning approaches based on behavioural genetics , to name a few .\n",
      "In a separate set of experiments , we tracked vsx1+ sister neurons from 48 to 96 hpf by embedding fish for imaging at 48 hpf , freeing from agarose after imaging , and re-embedding at 96 hpf .\n",
      "med .\n",
      "nlm .\n",
      "They are identified as ( i ) the Gaussian variability in response around the target value ( target response proportion ) , ( ii ) Gaussian variability in response around the non-target value , that is , mistakenly reporting feature of other presented items ( swap error ) , and ( iii ) random responses ( uniform response proportion ) ( Bays et al .\n",
      ", 1987 ) , rabbit anti-CD4 ( Novus Biologicals , 1:500 ) , mouse anti-REPO ( Developmental Studies Hybridoma Bank 8D12 , 1:5 ) , rabbit anti-DsRed ( Clontech #632496 , 1:200 ) , rat anti-HA ( Roche 3F10 , 1:100 ) , mouse anti-CaMPARI-red ( Absolute antibody 4F6 , 1:1000 ) , goat anti-HRP-Cy3 ( Jackson ImmunoResearch , 1:100 ) , goat anti-rat Alexa488 ( Invitrogen , 1:500 ) , goat anti-chicken Alexa488 ( Invitrogen , 1:500 ) , goat anti-mouse Alexa546 ( Invitrogen , 1:500 ) , goat anti-rabbit Alexa546 ( Invitrogen , 1:500 ) , goat anti-rat Alexa633 ( Invitrogen , 1:500 ) , and goat anti-rabbit Alexa633 ( Invitrogen , 1:500 ) .\n",
      "ac .\n",
      "The CCG at time lag τ is the correlation between the velocities of the target vt ( target ) at time t and the velocity of the response vt−τ ( response ) at time ( t−τ ) : ( 9 ) ρ ( τ ) =cov ( vt ( target ) , vt−τ ( response ) ) s ( vt ( target ) ) ⋅s ( vt−τ ( response ) ) , where cov⁢ ( x , y ) and s⁢ ( x ) are the covariance and standard deviation across all time steps and trials .\n",
      "Performing the change of variables using the derivative dξn+1dpn+1=σ22pn+1μ−2pn+12μ>0 yields the transfer function ( 16 ) fp ( pn+1|pn ) =12μpn+1 ( 1−pn+1 ) 2πδtσ[pnexp⁡{−12σ2δt ( σ22μln⁡ ( pn−1 ) pn+1pn ( pn+1−1 ) −δtμ ) 2}+ ( 1−pn ) exp⁡{−12σ2δt ( σ22μln⁡ ( pn−1 ) pn+1pn ( pn+1−1 ) +δtμ ) 2}] .\n",
      "The sediment samples were extracted using 5 ml of hexane: acetone ( 3:1 v/v ) , vortexed for 5 min , followed by ultrasonication for 15 min and centrifugation for 3 min at 5000 rpm .\n",
      "org/10 .\n",
      "ac .\n",
      "6 cm ) .\n",
      "Standard rodent diet and acidified tap water were provided ad libitum .\n",
      ", 2014; van der Hoorn et al .\n",
      "org/ ) .\n",
      "1 260/280 nm ratio .\n",
      "dna .\n",
      "4 mm voxel size , echo time ( TE ) = 20 ms , repetition time ( TR ) = 1 .\n",
      "nves-k !\n",
      "nov .\n",
      "org/ ) .\n",
      "01 ml/g ) .\n",
      "zip .\n",
      "We then modeled the bladder filling and voiding over a 6 hr duty cycle with different states: filling bladder ( 0 to 2 hr , 0% to 10% strain ) , filled bladder ( 2 to 4 hr , 10% strain ) , voiding bladder over a period of 2 min ( 4 hr to 4:02 hr , 10% to 0% strain ) , and voided bladder ( 4:02 hr to 6 hr , 0% strain ) ( Figure 1D ) .\n",
      "ac .\n",
      "nlm .\n",
      "The VPA treatment group received 200 mg/kg VPA in 15% DMSO and 85% saline , whereas the vehicle control group received the same volume of 15% DMSO and 85% saline without VPA .\n",
      ", 2011 ) , plated on LB agar containing ampicillin ( 100 µg/ml ) and kanamycin ( 50 µg/ml ) , and incubated for 16 hr at 37°C .\n",
      "7] v[+t1 .\n",
      "com/ .\n",
      "ed .\n",
      "2% glucose .\n",
      "6% w/v ) .\n",
      "soe .\n",
      "Blots were washed three times , 5 min each in PBST at room temperature with agitation before being incubated in 1:100 LARP1 mouse monoclonal ( Santa Cruz ) primary antibody in 2% BSA ( Research Products International A30075 ) in PBST overnight at 4 °C with agitation with 1:5000 beta actin mouse monoclonal antibody ( Sigma A5441 ) in 2% milk powder in PBST for 2 hr at room temperature with agitation or with 1:10 , 000 H3 mouse monoclonal antibody ( Abcam ab10799 ) in 2% milk powder in PBST for 2 hr at room temperature with agitation .\n",
      "org/10 .\n",
      "5 mg/kg docetaxel ( days 7 , 9 , 11 post-implantation ) and 1 dose of 10 mg/kg docetaxel ( day 14 post-implantation ) .\n",
      "nlm .\n",
      "com ) .\n",
      "io .\n",
      "org ) .\n",
      "ac .\n",
      "2% , odds ratio = 5 .\n",
      "0076 ) , albeit weaker ( R2=0 .\n",
      "New experimental condition terms were developed with a prefix of ‘delivery mechanism’: for example , ‘delivery mechanism: agrobacterium , ’ ‘delivery mechanism: heterologous organism , ’ and ‘delivery mechanism: pathogen inoculation .\n",
      "5 ng DNA .\n",
      "cbu .\n",
      "SD1 and SD2 form the outer domain , and SD3 and SD4 form the inner domain ( Figure 1 ) .\n",
      "umn .\n",
      "coli cell envelope .\n",
      "csc .\n",
      "To specify the Bayesian logistic regression ( BLR ) models for the probability of owning a mobile phone , we modeled phone ownership yij of individual i ∈ ( 1 , nj ) in country j ∈ ( 1 , 33 ) as a Bernoulli variable with probability θij : ( 1 ) yij|θij∼Bern ( θij ) The probability of owning a mobile phone , P ( yij=1 ) =θij , was then modeled using the logit-link function: ( 2 ) logit ( θij ) =β0+β1x1ij where β0 is the population-level intercept , and β1 is a regression coefficient that quantifies the influence of predictor variable x1ij ; we used a separate BLR model for each of the five variables .\n",
      "ac .\n"
     ]
    }
   ],
   "source": [
    "# write to a file with each summary in a new line\n",
    "outF = open('summary.txt',\"w\")\n",
    "for line in finalarr:\n",
    "  print(line)\n",
    "  outF.write(line)\n",
    "  outF.write(\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5×10–5 bar–1 ) .\n",
      "For example , wrongly rejecting a food item might see an animal missing a meal , but wrongly accepting a food item could see an animal ingesting poison .\n",
      "ac .\n",
      "Hosts were only included in these analyses if they had >35 samples in either the juvenile and prime age class or prime age and older adult class ( no host had >35 samples in all three age classes; 13 hosts were included in the juvenile and matched prime age adult groups; a separate set of 13 hosts were included in the older adult and matched prime age adult groups ) .\n",
      "doi .\n",
      "25 ng/µl .\n",
      "ilmn12 .\n",
      ", chr17q21 locus ) .\n",
      "Fertilizers were applied three times per year , namely once just before sowing ( 50 kg/ha N , 85 kg/ha P , 50 kg/ha K ) , once when wheat was at the tillering stage ( 50 kg/ha N , 85 kg/ha P , 50 kg/ha K ) , and once when wheat was flowering ( 20 kg/ha N , 34 kg/ha P , 20 kg/ha K ) .\n",
      "Then two choice targets appeared at 1 of 3 different pairs of positions ( 140 mm to the left and right of the central fixation target and either at the same height as the central fixation target or 35 mm below or 35 mm above it ) .\n",
      "at room temp .\n",
      ", 2009; van Geel et al .\n",
      "5 ) at a dose of 35 mg/kg for day-1 , and 35 mg/kg for day-2 at the 5th week .\n",
      "Turn angle ( α ) : the angle between the line passing through the CoM and the tip of the snout at the onset of stage 1 , and the line passing through the CoM at the onset of stage 1 and the CoM at the end of stage 2 .\n",
      "io .\n",
      "077 g/ml , cat .\n",
      "825 Å/pixel .\n",
      "org ) .\n",
      "5 mm .\n",
      "Figure 2 , Figure 2—source data 1 , Figure 2—source data 2 , Figure 2—source data 3 , Figure 2—source data 4 , Figure 3 , Figure 3—source data 1 , Figure 3—source data 2 , Figure 3—source data 3 , and Figure 3—source data 4 show the summarized effect estimates .\n",
      "1 mL/min on the following gradient: 0% B for 0–2 min , linear increase 0–35% B in 2–60 min , 35–85% B in 60–62 min , maintain at 85% B until 68 min , linear decrease to 0% in 68–70 min , and finally equilibrated the system at 0% B until 85 min .\n",
      "2 mg/ml k-casein ) .\n",
      "For mapping between euchromatin and heterochromatin ( Figure 3B ) , the following limits of euchromatin were used: chr = \"2L\" & start <21400000; chr = \"2R\" & start >5970000; chr = \"3L\" & start <22270000 chr = \"3R\" & start >4191000; chr == \"X\" & start <21400000 75 b .\n",
      "ed .\n",
      "edu ) .\n",
      "CD11B+ gating quantifies the myeloid compartment .\n",
      "msu .\n",
      "0 eq ) .\n",
      "pct=0 .\n",
      "mulitqtl .\n",
      "5 µm step size .\n",
      "jsp ) .\n",
      "ac .\n",
      "4th , 5th , etc . )\n",
      "ac .\n",
      "After a 10 min wash with WB transfer buffer supplied with 1 mM EDTA , followed by a 10 min wash with transfer buffer without EDTA .\n",
      "io; probability > 0 .\n",
      "2% Triton X-100 and 1% BSA ) , using the following antibodies and dilutions: anti-PDGFRβ/CD140b ( dilution 1:200 , clone APB5 , eBioscience ) +anti rat AF647 ( dilution 1:200 , Jackson IR ) ; anti-CD11c AF647 , ( dilution 1:100 , clone N418 , Biolegend ) ; anti-GFP ( dilution 1:100 , rabbit polyclonal , A11122 Thermo Fisher Scientific ) +anti-rabbit AF 488 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-CD31/PECAM-1 ( dilution 1:300 , goat polyclonal , AF3628 R&D Systems ) +anti goat AF546 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-Heparan Sulfate ( dilution 1:50 , clone F58-10E4 , Amsbio ) +anti IgM conjugated to APC ( dilution 100 , clone II/41 , Thermo Fisher Scientific ) ; anti-LYVE-1 ( dilution 1:300 , rabbit polyclonal , Novus Biologicals ) +anti-rabbit AF647 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-Collagen type IV ( dilution 1:100 , rabbit polyclonal , Abcam ) +anti-rabbit AF488 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-Laminin ( dilution 1:300 , rabbit polyclonal , Sigma-Aldrich ) +anti-rabbit AF488 ( dilution 1:200 , Thermo Fisher Scientific ) ; anti-CD54/ICAM1 ( dilution 1:100 , clone YN1/1 .\n",
      "vcf .\n",
      "5 mm ) .\n",
      "89 log 10 copies/ml , p=0 .\n",
      "dna .\n",
      "Among the 18 infants , 11 performed newborn stepping , 15 performed stepping at 3 months old , 15 performed newborn kicking , 15 performed kicking at 3 months old , and 15 performed walking as toddlers ( Table 2 ) .\n",
      "ox .\n",
      "1 g bovine serum albumin .\n",
      "org/ ) .\n",
      "org/ ) .\n",
      "ac .\n",
      "Optimal E was selected by measuring TE as follows: ( 2 ) TE=1T∑t=1Tlog⁡ ( p ( xt+tp | yt , xt , xt−τ , ⋯ , xt− ( E−1 ) τ ) p ( xt+tp | yt , xt , xt−τ , ⋯ , xt− ( ER−1 ) τ ) ) , where ER ( <E ) is the optimal embedding dimension of lower dimensional models .\n",
      ", 2019 , table 1 ) .\n",
      "815 mm/pixel ) .\n",
      "2 pixel .\n",
      "( 17 ) cos⁡ ( 2β ) = ( a~2−b~2a~2+b~2 ) 1+25η−G′′/ωη1K ( a~2+b~22a~b~ ) 21+25η−G′′/ωη1K ( a~2−b~22a~b~ ) 2 ( corresponding to Equation 80 in Roscoe , 1967 ) .\n",
      "mit .\n",
      "4% Agar gel with the WT allele producing a 1868 bp amplicon and the floxed allele producing a 2142 bp amplicon .\n",
      "com/ ) .\n",
      "coli .\n",
      "05 cutoff .\n",
      "py .\n",
      ", politics , health , science , etc . )\n",
      "7 nm .\n",
      "nlm .\n",
      "5 kV ) mode electrospray ionization ( capillary temperature 300 °C; sheath gas 50; Aux gas 20; sweep gas 2; probe temp 120 °C ) .\n",
      "twofolds of 1% would remarkably differ from twofolds of 10% ) .\n",
      "nitrc .\n",
      "ucd .\n",
      "org/10 .\n",
      ", 2004 ) ; tissue 1 = cortical grey matter , tissue 2 = sub-cortical grey matter , tissue 3 = white matter , tissue 4 = CSF , and tissue 5 = pathological tissue .\n",
      "1% Thioflavin-S in 70% ethanol 25 min , rinsed in 80% ethanol for 5 min and in distilled water for 5 min .\n",
      "We used the primary antibodies: anti-CD44 ( clone IM7 ) , anti-MHC class I ( clone 28-8-6 ) ( both from BioLegend ) , anti-β-catenin ( clone D2U8Y ) , anti-phosphorylated STAT3 ( clone D3A7 ) , anti-active integrin 1β ( polyclonal ) , anti-Granzyme B ( clone 5A1E ) ( all from Cell Signaling Technology ) , anti-integrin 1β ( clone 9EG7 ) , anti-E-cadherin ( clone 36 ) ( both from BD Biosciences ) , anti-TRP2 ( clone C-9 , Santa Cruz Biotechnology ) , anti-gp100 ( clone EP4863 Galon et al .\n",
      "001 , ηp2p2 .\n",
      "org/10 .\n",
      "4 ) , and cut into multiple 1 mm × 1 mm strips in fluid; several strips were simultaneously cut into 1 mm × 1 mm × 1 mm blocks for TEM .\n",
      "/ml ) .\n",
      "2 cm2; slice number = 20; slice thickness = 1 mm; repetition number = 6 .\n",
      "et al .\n",
      "org/10 .\n",
      "Enrichment analysis identified Gene Ontology ( GO ) biological processes linked to the cardiovascular system including striated muscle cell development , muscle cell development , muscle system process , cardiac muscle contraction , heart contraction , heart process , contraction muscle cell development , and muscle tissue progress among others ( Figure 6b ) .\n",
      "The surface function of Imaris was used to segment the wing pouch and the surface volume was calculated by the software .\n",
      "yield ( i .\n",
      "fa .\n",
      "Time-resolved oxygen polarography was performed using a custom-made centrifugable static ring-disk electrode assembly of a bare platinum working electrode and silver-ring counter electrode , as previously described ( Dilbeck et al .\n",
      "nlm .\n",
      "msu .\n",
      "Sequencing was performed using HiSeq 4000 ( Illumina ) at IGM Genomics Center , University of California , San Diego , with the following read length: Read 1 , 28 bp , including 16 bp cell barcode and 12 bp unique molecular identifier ( UMI ) ; Read 2 , 98 bp transcript insert; i7 sample index , 8 bp .\n",
      "uth .\n",
      "The ion source settings were capillary voltage , 3500 V; nozzle voltage , 2000 V; fragmentor voltage , 180 V; drying gas and sheath gas temperature , 350°C; drying gas flow , 10 L/min; sheath gas flow , 11 L/min; nebulizer pressure , 60 lb/in2 .\n",
      "5 nm .\n",
      "Various companies now sell personalised diets based on nutrigenomics or personalised exercise plans based on sportomics; others promise personalised learning approaches based on behavioural genetics , to name a few .\n",
      "In a separate set of experiments , we tracked vsx1+ sister neurons from 48 to 96 hpf by embedding fish for imaging at 48 hpf , freeing from agarose after imaging , and re-embedding at 96 hpf .\n",
      "med .\n",
      "nlm .\n",
      "They are identified as ( i ) the Gaussian variability in response around the target value ( target response proportion ) , ( ii ) Gaussian variability in response around the non-target value , that is , mistakenly reporting feature of other presented items ( swap error ) , and ( iii ) random responses ( uniform response proportion ) ( Bays et al .\n",
      ", 1987 ) , rabbit anti-CD4 ( Novus Biologicals , 1:500 ) , mouse anti-REPO ( Developmental Studies Hybridoma Bank 8D12 , 1:5 ) , rabbit anti-DsRed ( Clontech #632496 , 1:200 ) , rat anti-HA ( Roche 3F10 , 1:100 ) , mouse anti-CaMPARI-red ( Absolute antibody 4F6 , 1:1000 ) , goat anti-HRP-Cy3 ( Jackson ImmunoResearch , 1:100 ) , goat anti-rat Alexa488 ( Invitrogen , 1:500 ) , goat anti-chicken Alexa488 ( Invitrogen , 1:500 ) , goat anti-mouse Alexa546 ( Invitrogen , 1:500 ) , goat anti-rabbit Alexa546 ( Invitrogen , 1:500 ) , goat anti-rat Alexa633 ( Invitrogen , 1:500 ) , and goat anti-rabbit Alexa633 ( Invitrogen , 1:500 ) .\n",
      "ac .\n",
      "The CCG at time lag τ is the correlation between the velocities of the target vt ( target ) at time t and the velocity of the response vt−τ ( response ) at time ( t−τ ) : ( 9 ) ρ ( τ ) =cov ( vt ( target ) , vt−τ ( response ) ) s ( vt ( target ) ) ⋅s ( vt−τ ( response ) ) , where cov⁢ ( x , y ) and s⁢ ( x ) are the covariance and standard deviation across all time steps and trials .\n",
      "Performing the change of variables using the derivative dξn+1dpn+1=σ22pn+1μ−2pn+12μ>0 yields the transfer function ( 16 ) fp ( pn+1|pn ) =12μpn+1 ( 1−pn+1 ) 2πδtσ[pnexp⁡{−12σ2δt ( σ22μln⁡ ( pn−1 ) pn+1pn ( pn+1−1 ) −δtμ ) 2}+ ( 1−pn ) exp⁡{−12σ2δt ( σ22μln⁡ ( pn−1 ) pn+1pn ( pn+1−1 ) +δtμ ) 2}] .\n",
      "The sediment samples were extracted using 5 ml of hexane: acetone ( 3:1 v/v ) , vortexed for 5 min , followed by ultrasonication for 15 min and centrifugation for 3 min at 5000 rpm .\n",
      "org/10 .\n",
      "ac .\n",
      "6 cm ) .\n",
      "Standard rodent diet and acidified tap water were provided ad libitum .\n",
      ", 2014; van der Hoorn et al .\n",
      "org/ ) .\n",
      "1 260/280 nm ratio .\n",
      "dna .\n",
      "4 mm voxel size , echo time ( TE ) = 20 ms , repetition time ( TR ) = 1 .\n",
      "nves-k !\n",
      "nov .\n",
      "org/ ) .\n",
      "01 ml/g ) .\n",
      "zip .\n",
      "We then modeled the bladder filling and voiding over a 6 hr duty cycle with different states: filling bladder ( 0 to 2 hr , 0% to 10% strain ) , filled bladder ( 2 to 4 hr , 10% strain ) , voiding bladder over a period of 2 min ( 4 hr to 4:02 hr , 10% to 0% strain ) , and voided bladder ( 4:02 hr to 6 hr , 0% strain ) ( Figure 1D ) .\n",
      "ac .\n",
      "nlm .\n",
      "The VPA treatment group received 200 mg/kg VPA in 15% DMSO and 85% saline , whereas the vehicle control group received the same volume of 15% DMSO and 85% saline without VPA .\n",
      ", 2011 ) , plated on LB agar containing ampicillin ( 100 µg/ml ) and kanamycin ( 50 µg/ml ) , and incubated for 16 hr at 37°C .\n",
      "7] v[+t1 .\n",
      "com/ .\n",
      "ed .\n",
      "2% glucose .\n",
      "6% w/v ) .\n",
      "soe .\n",
      "Blots were washed three times , 5 min each in PBST at room temperature with agitation before being incubated in 1:100 LARP1 mouse monoclonal ( Santa Cruz ) primary antibody in 2% BSA ( Research Products International A30075 ) in PBST overnight at 4 °C with agitation with 1:5000 beta actin mouse monoclonal antibody ( Sigma A5441 ) in 2% milk powder in PBST for 2 hr at room temperature with agitation or with 1:10 , 000 H3 mouse monoclonal antibody ( Abcam ab10799 ) in 2% milk powder in PBST for 2 hr at room temperature with agitation .\n",
      "org/10 .\n",
      "5 mg/kg docetaxel ( days 7 , 9 , 11 post-implantation ) and 1 dose of 10 mg/kg docetaxel ( day 14 post-implantation ) .\n",
      "nlm .\n",
      "com ) .\n",
      "io .\n",
      "org ) .\n",
      "ac .\n",
      "2% , odds ratio = 5 .\n",
      "0076 ) , albeit weaker ( R2=0 .\n",
      "New experimental condition terms were developed with a prefix of ‘delivery mechanism’: for example , ‘delivery mechanism: agrobacterium , ’ ‘delivery mechanism: heterologous organism , ’ and ‘delivery mechanism: pathogen inoculation .\n",
      "5 ng DNA .\n",
      "cbu .\n",
      "SD1 and SD2 form the outer domain , and SD3 and SD4 form the inner domain ( Figure 1 ) .\n",
      "umn .\n",
      "coli cell envelope .\n",
      "csc .\n",
      "To specify the Bayesian logistic regression ( BLR ) models for the probability of owning a mobile phone , we modeled phone ownership yij of individual i ∈ ( 1 , nj ) in country j ∈ ( 1 , 33 ) as a Bernoulli variable with probability θij : ( 1 ) yij|θij∼Bern ( θij ) The probability of owning a mobile phone , P ( yij=1 ) =θij , was then modeled using the logit-link function: ( 2 ) logit ( θij ) =β0+β1x1ij where β0 is the population-level intercept , and β1 is a regression coefficient that quantifies the influence of predictor variable x1ij ; we used a separate BLR model for each of the five variables .\n",
      "ac .\n"
     ]
    }
   ],
   "source": [
    "# write to a file with each summary in a new line\n",
    "outF = open('summary.txt',\"w\")\n",
    "for line in finalarr:\n",
    "  print(line)\n",
    "  outF.write(line)\n",
    "  outF.write(\"\\n\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
